### ISSUE TITLE
odoo_repository : remove dependency on scanner

### ISSUE THREAD

**Issue by hparfr** [OCA/module-composition-analysis#55]
Hi,

I plan to move the lib and  scan buttons out of odoo_repository.
It will be a new module like odoo_repository_scanner (or what ever your prefer). 

This module will depend of odoo_repository module.

**Comment by sebalix** [comment]
A step I would like to take regarding this, is get rid of the overrides that includes `env` parameter in the scanner, and replace this by HTTP calls on Odoo (so the scanner would need the Odoo URL and credentials to authenticate).

This will make things clear about the interface between the scanner(s) and Odoo, and will expose a clear endpoint where to push data (that could be used by any other tools).

Once it is done, we could easily extract the scanner in a separate lib.

**Comment by hparfr** [comment]
good; It's low priority for me, but i think it leads to a better design.

For the record; I plan to have this setup:

- an instance with the current modules
 + use the scanner to scan public repositories (oca / shopinvader / akretion), and some private repository (but not project's private repo)
 + and a new module to expose an HTTP endpoint 

- a module installed in client's production. this module will push data to the HTTP API endpoint (module list of installed modules, list of PR used...)


At the moment, I don't plan to scan the projects' private repositories, but to push data from running instance of odoo.

**Comment by sebalix** [comment]
I aim to get a "phonehome" module as well to push installed modules from a running project, that could be extended as well to push other metrics like info about mail.template (edited+translated), manually created or studio views... this helps to better estimate a migration by functional people.

EDIT: Ideally, we should be able to configure any metrics/placeholders on "Odoo Repository" instance, and free to customer instances to push data regarding these metrics, that will be stored in "Odoo Repository" in text/JSON fields. Such data will have to be rendered in the migration report too.
